src/d3d9/d3d9_device.cpp

```diff
@@ -3644,48 +3644,47 @@ namespace dxvk {
 
     auto& state = m_state.samplerStates;
 
-    bool changed = state[StateSampler][Type] != Value;
+    if (state[StateSampler][Type] == Value)
```

Comment:
Why don't you just make the early return unlikely instead of wrapping everything in the if block?